libs/agno/agno/storage/dynamodb.py

```diff
@@ -675,3 +675,93 @@ def deserialize_value(value):
                 return value
 
         return {k: deserialize_value(v) for k, v in item.items()}
+
+    def get_workflow_run_status(self, session_id: str, run_id: str) -> Optional[Dict[str, Any]]:
+        """Fast retrieval of workflow run status from DynamoDB"""
+        try:
+            # Get only the specific fields we need for status checking
+            response = self.table.get_item(
+                Key={"session_id": session_id},
+                ProjectionExpression="current_run_id, current_run_status, updated_at, runs",
+            )
+
+            item = response.get("Item")
+            if item is None:
+                return None
+
+            # Deserialize the item to handle Decimal conversions
+            item = self._deserialize_item(item)
+
+            # Check if this session has the run_id we're looking for
+            current_run_id = item.get("current_run_id")
+            if current_run_id == run_id:
+                return {
+                    "run_id": current_run_id,
+                    "status": item.get("current_run_status"),
+                    "updated_at": item.get("updated_at", int(time.time())),
+                }
+
+            # If current_run_id doesn't match, check in the runs array
+            runs = item.get("runs", [])
+            for run in runs:
+                if isinstance(run, dict) and run.get("run_id") == run_id:
+                    return {
+                        "run_id": run_id,
+                        "status": run.get("status"),
+                        "updated_at": run.get("updated_at", int(time.time())),
+                    }
+
+        except Exception as e:
+            logger.error(f"Failed to get workflow run status: {e}")
+        return None
+
+    def update_workflow_run_status(self, session_id: str, run_id: str, status: str) -> bool:
```

Comment:
Similarly this should just store an updated run response. You could actually just do the "upsert"